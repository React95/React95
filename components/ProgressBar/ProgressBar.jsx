import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';

const Wrapper = styled.div`
  width: ${props => props.width}px;
  height: 20px;

  position: relative;

  text-align: center;
`;

const WhiteBar = styled.div`
  width: ${props => props.width}px;
  height: 20px;
  line-height: 20px;

  border-left: 1px solid #868a8e;
  border-top: 1px solid #868a8e;

  background: #fff;
  color: #000;

  box-shadow: inset -1px -1px 0 0 #c3c7cb, inset 1px 1px 0 0 #000000,
    0.5px 0.5px 0 0.5px #ffffff;
`;

const Container = styled.div`
  width: ${props => props.percent}%;

  position: absolute;
  top: 0;
  left: 0;

  overflow: hidden;
`;

const BlueBar = styled.div`
  width: ${props => props.width}px;
  height: 17px;
  line-height: 18px;

  margin-left: 2px;
  margin-top: 2px;

  background: #000e7a;
  color: #fff;
`;

const ProgressBar = ({ width, percent }) => (
  <Wrapper width={width}>
    <WhiteBar width={width}>
      {percent}
%
    </WhiteBar>
    <Container percent={percent}>
      <BlueBar width={width}>
        {percent}
%
      </BlueBar>
    </Container>
  </Wrapper>
);

ProgressBar.propTypes = {
  width: PropTypes.number,
  percent: PropTypes.number,
};

ProgressBar.defaultProps = {
  width: 150,
  percent: 0,
};

export default ProgressBar;
