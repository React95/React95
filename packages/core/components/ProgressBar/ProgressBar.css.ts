import { style } from '@vanilla-extract/css';
import { contract } from '../themes/contract.css';

export const wrapper = style({
  height: contract.space[20],
  position: 'relative',
  textAlign: 'center',
});

export const whiteBar = style({
  height: contract.space[20],
  lineHeight: contract.space[20],
  borderLeft: contract.space[1],
  borderLeftStyle: 'solid',
  borderLeftColor: contract.colors.borderDark,
  borderTop: contract.space[1],
  borderTopStyle: 'solid',
  borderTopColor: contract.colors.borderDark,
  backgroundColor: contract.colors.inputBackground,
  color: contract.colors.materialText,
  boxShadow: contract.shadows.input,
});

export const container = style({
  position: 'absolute',
  top: '0',
  left: '0',
  overflow: 'hidden',
});

export const progress = style({
  height: contract.space[17],
  lineHeight: contract.space[18],
  marginLeft: contract.space[2],
  marginTop: contract.space[2],
  backgroundColor: contract.colors.progress,
  color: contract.colors.materialTextInvert,
});
