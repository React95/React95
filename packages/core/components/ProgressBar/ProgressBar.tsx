import React, { forwardRef } from 'react';
import type {
  ElementType,
  ReactElement,
  ElementRef,
  ForwardedRef,
} from 'react';
import cn from 'classnames';

import { Frame, FrameProps } from '../Frame/Frame';
import * as styles from './ProgressBar.css';

export type ProgressBarProps<TAs extends ElementType> = {
  percent?: number;
} & FrameProps<TAs>;

export const ProgressBar = forwardRef<HTMLDivElement, ProgressBarProps<'div'>>(
  ({ width = '150px', percent = 0, ...rest }, ref) => (
    <Frame
      {...rest}
      width={width}
      className={cn(styles.wrapper, rest.className)}
      ref={ref}
    >
      <Frame className={styles.whiteBar} width={width}>{`${percent}%`}</Frame>
      <Frame className={styles.container} width={`${percent}%`}>
        <Frame className={styles.progress} width={width}>{`${percent}%`}</Frame>
      </Frame>
    </Frame>
  ),
) as <TAs extends ElementType = 'div'>(
  props: ProgressBarProps<TAs> & { ref?: ForwardedRef<ElementRef<TAs>> },
) => ReactElement;
