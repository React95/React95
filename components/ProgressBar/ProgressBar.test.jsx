import React from 'react';
import ProgressBar from './ProgressBar';
import { render } from 'react-testing-library';

describe('<ProgressBar />', () => {
  describe('Snapshots', () => {
    it('should match snapshot', () => {
      const { container } = render(<ProgressBar width={300} percent={50} />);
      expect(container).toMatchSnapshot();
    });
  });
});
