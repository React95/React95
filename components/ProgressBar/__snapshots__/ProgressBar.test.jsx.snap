// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> Snapshots should match snapshot 1`] = `
.c0 {
  width: 300px;
  height: 20px;
  position: relative;
  text-align: center;
}

.c1 {
  width: 300px;
  height: 20px;
  line-height: 20px;
  border-left: 1px solid #868a8e;
  border-top: 1px solid #868a8e;
  background: #fff;
  color: #000;
  box-shadow: inset -1px -1px 0 0 #c3c7cb,inset 1px 1px 0 0 #000000,0.5px 0.5px 0 0.5px #ffffff;
}

.c2 {
  width: 50%;
  position: absolute;
  top: 0;
  left: 0;
  overflow: hidden;
}

.c3 {
  width: 300px;
  height: 17px;
  line-height: 18px;
  margin-left: 2px;
  margin-top: 2px;
  background: #000e7a;
  color: #fff;
}

<div>
  <div
    class="c0"
    width="300"
  >
    <div
      class="c1"
      width="300"
    >
      50
      %
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
        width="300"
      >
        50
        %
      </div>
    </div>
  </div>
</div>
`;
